static_frame/core/store_zip.py

```diff
@@ -106,36 +153,60 @@ def read_many(self,
         multiprocess: bool = config_map.default.read_max_workers is not None
         constructor: FrameConstructor = self._container_type_to_constructor(container_type)
 
-        def gen() -> tp.Iterable[tp.Union[PayloadBytesToFrame, Frame]]:
+        if not multiprocess:
+            yield from self._read_many_single_thread(
+                    labels=labels,
+                    config_map=config_map,
+                    constructor=constructor,
+                    container_type=container_type,
+            )
+            return
+
+        # To ensure race-conditions don't happen, and to simplifiy the code
+        # we lock in the state of our weak_cache for the rest of the function
+        strong_cache = dict(self._weak_cache)
```

Comment:
I have some concern for the performance implications of this: when `labels` is small and the size of `self._weak_cache` is large, we might suffer a lot of overheard for each of these temporary dictionaries.

Reply:
Thanks for noticing this. I will re-write this to be more efficient